stwcs/updatewcs/det2im.py

```diff
@@ -115,39 +115,25 @@ def addSciExtKw(cls, hdr, wdvarr_ver=None, d2im_extname=None, error_val=0.0):
         Adds kw to sci extension to define WCSDVARR lookup table extensions
 
         """
-        if d2im_extname == 'DX':
-            j = 1
-        else:
-            j = 2
-
-        d2imerror = 'D2IMERR%s' % j
-        d2imdis = 'D2IMDIS%s' % j
-        d2imext = 'D2IM%s.' % j + 'EXTVER'
-        d2imnaxes = 'D2IM%s.' % j + 'NAXES'
-        d2imaxis1 = 'D2IM%s.' % j + 'AXIS.1'
-        d2imaxis2 = 'D2IM%s.' % j + 'AXIS.2'
-        keys = [d2imerror, d2imdis, d2imext, d2imnaxes, d2imaxis1, d2imaxis2]
-        values = {d2imerror: error_val,
-                  d2imdis: 'Lookup',
-                  d2imext: wdvarr_ver,
-                  d2imnaxes: 2,
-                  d2imaxis1: 1,
-                  d2imaxis2: 2}
-
-        comments = {d2imerror: 'Maximum error of NPOL correction for axis %s' % j,
-                    d2imdis: 'Detector to image correction type',
-                    d2imext: 'Version number of WCSDVARR extension containing d2im lookup table',
-                    d2imnaxes: 'Number of independent variables in d2im function',
-                    d2imaxis1: 'Axis number of the jth independent variable in a d2im function',
-                    d2imaxis2: 'Axis number of the jth independent variable in a d2im function'
-                    }
+        j = 1 if d2im_extname == 'DX' else 2
+
+        jth = {1: '1st', 2: '2nd', 3: '3rd'}.get(j, f'{j}th')
+
+        d2im = [
+            (f'D2IMERR{j:1d}', error_val, f'Maximum error of D2IM correction for axis {j:d}'),
+            (f'D2IMDIS{j:1d}', 'Lookup', 'Detector to image correction type'),
+            (f'D2IM{j:1d}.EXTVER', wdvarr_ver, 'Version number of WCSDVARR extension'),
+            (f'D2IM{j:1d}.NAXES', 2, 'Number of independent variables in D2IM function'),
+            (f'D2IM{j:1d}.AXIS.1', 1, 'Axis number of the {jth} variable in a D2IM function'),
+            (f'D2IM{j:1d}.AXIS.2', 2, 'Axis number of the {jth} variable in a D2IM function'),
```

Comment:
I would suggest hard coding the axis number instead of using `{jth}` because `j` here means 1st or 2nd axis for either of the `DX`, `DY` extensions, i.e. `jth` in the comment is different from `j` in the code.